### PR TITLE
Add first try at a NED data source.

### PR DESCRIPTION
**Pull request by zerebubuth**
This is the first attempt at adding a NED data source. There are some outstanding issues, which I've punted to #2 and #3; respectively, that at least one file fails to download in an interesting way, and that NED contains many overlapping data files and it would be best to not download all of them (at the moment, I've set it to only download the "topobathymetry" ones, but they don't seem to have full coverage by themselves.

Connects to #1.

@rmarianski could you review, please?

### CODE REVIEW

File: joerd/source/ned.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+import traceback
+import subprocess
+import glob
+from osgeo import gdal
+import urllib2
+import shutil
+
+
+NED_FTP_SERVER = 'rockyftp.cr.usgs.gov'
+NED_BASE_PATH = 'vdelivery/Datasets/Staged/NED/19/IMG'
```
Comment (rmarianski) [inline joerd/source/ned.py]: Probably want to move this into a config file eventually.

File: joerd/source/ned.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        logger.info("FTP: Fetching %r" % url)
+
+        with closing(urllib2.urlopen(url)) as req:
+            shutil.copyfileobj(req, tmp)
+
+        tmp.flush()
+
+        with zipfile.ZipFile(tmp.name, 'r') as zfile:
+            zfile.extract(img_name, base_dir)
+            zfile.extract(img_name + ".aux.xml", base_dir)
```
Comment (rmarianski) [inline joerd/source/ned.py]: Can we have the zip file take the stream directly? Or does it need the whole file in  advance?

File: joerd/source/ned.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        logger.info("Creating VRT.")
+
+        files = []
+        for f in glob.glob(os.path.join(self.base_dir, '*.img')):
+            bbox = _ned_parse_filename(os.path.split(f)[1])
+            if bbox and self._intersects(bbox):
+                files.append(f)
+
+        args = ["gdalbuildvrt", "-q", self.vrt_file()] + files
+        status = subprocess.call(args)
```
Comment (rmarianski) [inline joerd/source/ned.py]: Is there a python interface for this? Or do we need to subprocess?

**Comment by rmarianski** [comment]
:+1:

File: joerd/source/ned.py (hunk 1 above)
Comment (zerebubuth) [inline joerd/source/ned.py]: Good point! Added #5 to track this.

File: joerd/source/ned.py (hunk 2 above)
Comment (zerebubuth) [inline joerd/source/ned.py]: Good question - I've added #6 to investigate this.

File: joerd/source/ned.py (hunk 3 above)
Comment (zerebubuth) [inline joerd/source/ned.py]: I couldn't find a Python interface to `gdalbuildvrt` itself, although there's functionality in the GDAL Python bindings to replicate its functionality. The question (#7) is how much there is in there that we'd need to reimplement. Perhaps there's an additional library for it?

**Pull request merged**